### DISCUSSION TITLE
Release 1.16

### DISCUSSION THREAD

**Discussion by webspiderteam** [webspiderteam/Bluetooth-DMM-For-Windows#17] [Announcements]
- Fix ValueF returns last value(MQTT ValueF now returns null if display has non numeric value)
- Menu optimization

<hr /><em>This discussion was created from the release <a href='https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/releases/tag/WPFRelease1.16'>Release 1.16</a>.</em>

**Comment by ludwich66** [comment]
Review 1.16

1.16-1 
Diode Mode JSON parsing error
"{ "Time":"04.08.2022 22:49:45", "Device":"ANENG V05B", "ValueS":" .0L ", "Range":"V", "Current":"", "Symbols":"[ "BUZZ","DIODE" ]", "ValueF":null }"

to 1.16-1 capsulation isn't working (https://jsonformatter.curiousconcept.com)
MQTT-String Error in part Symbols - this kind of capsulation works (on/off True/False 1/0)
{ "Time":"04.08.2022 23:36:00", "Device":"ANENG V05B", "ValueS":" .0L ", "Range":"V", "Current":"", "Symbols":{"BUZZ":True, "DIODE": True}, "ValueF":null }


1.16-2 Problems with float, decimalseperator "comma"/"dot" changed

Interesting Message from Node Red if DMM goes sleeping
Status: "Disconnected"
MAC: "FC_58_FA_77_0B_33"

Good night!

**Reply by webspiderteam** [reply]
> Interesting Message from Node Red if DMM goes sleeping
> Status: "Disconnected"
> MAC: "FC_58_FA_77_0B_33"

When sleep mode Bluetooth connection disconnects

**Reply by webspiderteam** [reply]
> 1.16-2 Problems with float, decimalseperator "comma"/"dot" changed

I tested and it always gives as dot as its suppose to be because programming languages uses dot for decimals and comas for seperating values { "Time":"04.08.2022 23:36:00", "Device":"ANENG V05B", "ValueS":" .0L ", "Range":"V", "Current":"", "Symbols":{"BUZZ":True, "DIODE": True}, "ValueF":null } like that also if you try to send fload with coma you will get parse error in json try it on https://jsonformatter.curiousconcept.com/

**Comment by webspiderteam** [comment]
{ "Time":"04.08.2022 22:49:45", "Device":"ANENG V05B", "ValueS":" .0L ", "Range":"V", "Current":"", "Symbols":[ "BUZZ","DIODE" ], "ValueF":null } I forgot to remove quotes now it's ok.
BTW try press Ctrl and hold and click menu then settings don't release Ctrl until click settings

**Comment by ludwich66** [comment]
Play with JSON formats ;-)

ValueS and Range   --> Range = mx, µx, nx, px, kΩ, MΩ, etc , x=unit
ValueF and Unit --> FloatingValue and BaseUnits A,V,Ω,F,°C,°F


JSON Sring:
`{ "Time":"05.08.2022 11:22:24", "Device":"ANENG V05B", Value:{"ValueS":"2.326", "Range":"MΩ"}, Value_SI:{"ValueF":2326000, Unit:"Ω"}, "Current":"", "Symbols":{"BUZZ":true, "DIODE":true}}`

and parsed view:
```
{
   "Time":"05.08.2022 11:22:24",
   "Device":"ANENG V05B",
   "Value":{
      "ValueS":"2.326",
      "Range":"MΩ"
   },
   "Value_SI":{
      "ValueF":2326000,
      "Unit":"Ω"
   },
   "Current":"",
   "Symbols":{
      "BUZZ":true,
      "DIODE":true
   }
}
```
collapsed:
![image](https://user-images.githubusercontent.com/12202733/183060478-499d41e9-2831-4c40-9674-167e57a4c5e2.png)
open:
![image](https://user-images.githubusercontent.com/12202733/183060397-112613fb-9020-4666-8d55-0cffe61e4f58.png)
---------------------------------------------
another with "null" and without ValueF
{ "Time":"05.08.2022 11:13:46", "Device":"ANENG V05B", Value:{"ValueS":" .0L ", "Range":"V"}, "Current":"", "Symbols":{"BUZZ":True,"DIODE":True}, "ValueF":null }

I hope you can convert dot/comma (;-) or change to englisch localisation)

have fun!

**Reply by webspiderteam** [reply]
> I hope you can convert dot/comma (;-) or change to englisch localisation)

you cant do it in json format. coma is seperator for each values 

{ "Time":"05.08.2022 11:13:46", "Device":"ANENG V05B", Value:{"ValueS":" .0L ", "ValueF": 1,05, "Range":"V"}, "Current":"", "Symbols":{"BUZZ":True,"DIODE":True} }   try to parse this you will understand.

Lets let them some coding. Everthing can't be ready for use for everyone. let them convert some value and some coding

**Comment by webspiderteam** [comment]
[Release.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/9267875/Release.zip)

**Comment by ludwich66** [comment]
Komma dosent work in JSON validator, i have changed colon to semicolon as separator tag but it doesent work too.

{ "Time":"05.08.2022 12:52:00"; "Device":"ANENG V05B"; "ValueF":61,7; "ValueS":" 61.7"; "Range":"kΩ"; "Current":""; "Symbols":[ ] }

**workaround** -> cover it as string first   "ValueF":"61.7" 
That way we can at least catch the JSON Validation error

--> And please ad "Unit":"basicunit"
then the pair with ValueF and Unit is complete.

THNX Ludwich

**Comment by ludwich66** [comment]
where is a good place to store the iconset?
[BT-DMM-ICONSET.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/9270578/BT-DMM-ICONSET.zip)
![image](https://user-images.githubusercontent.com/12202733/183128864-61bf4e8b-ceff-4aca-a137-138bd8aaf0e4.png)

**Reply by webspiderteam** [reply]
Under BluetoothLeExplorer

**Reply by ludwich66** [reply]
Please put the icons down yourself initially, even if it's just for "picking up". I can then simply upload changes to the (svg/jpg-preview) or new symbols via pull request.
greetings ludwich
